doc/design/OrionHDFSSink.md

```diff
@@ -24,27 +24,37 @@ Independently of the data generator, NGSI context data is always [transformed](f
 2. According to the [naming conventions](naming_convetions.md), a folder called `/user/<hdfs_userame>/<fiware-service>/<fiware-servicePath>/<destination>` is created (if not existing yet), where `<hdfs_username>` is a configuration parameter, and `<fiware_service>`, `<fiware-servicePath>` and `<destination>` values are got from the event headers.
 3. The context responses/entities within the container are iterated, and a file called `<destination>.txt` is created (if not yet existing), where `<destination>` value is got from the event headers.
-4. The context attributes within each context response/entity are iterated, and a new Json line (or lines) is appended to the current file. The format for this append depends on the configured persistence mode:
-    * `row`: A Json line is added for each notified context attribute. This kind of line will always contain 8 fields:
+4. The context attributes within each context response/entity are iterated, and a one or more lines are appended to the current file. The format for this append depends on the configured persistence mode:
+    * `json-row`: A JSON line is added for each notified context attribute. This kind of line will always contain 8 fields:
         * `recvTimeTs`: UTC timestamp expressed in miliseconds.
         * `recvTime`: UTC timestamp in human-redable format ([ISO 6801](http://en.wikipedia.org/wiki/ISO_8601)).
         * `entityId`: Notified entity identifier.
         * `entityType`: Notified entity type.
         * `attrName`: Notified attribute name.
         * `attrType`: Notified attribute type.
-        * `attrValue`: In its simplest form, this value is just a string, but since Orion 0.11.0 it can be Json object or Json array.
-        * `attrMd`: It contains a string serialization of the metadata array for the attribute in Json (if the attribute hasn't metadata, an empty array `[]` is inserted).
-    * `column`: A single Json line is added for all the notified context attributes. This kind of line will contain two fields per each entity's attribute (one for the value, called `<attrName>`, and other for the metadata, called `<attrName>_md`), plus an additional field about the reception time of the data (`recvTime`).
+        * `attrValue`: In its simplest form, this value is just a string, but since Orion 0.11.0 it can be JSON object or JSON array.
+        * `attrMd`: It contains a string serialization of the metadata array for the attribute in JSON (if the attribute hasn't metadata, an empty array `[]` is inserted).
+    * `json-column`: A single JSON line is added for all the notified context attributes. This kind of line will contain two fields per each entity's attribute (one for the value, called `<attrName>`, and another for the metadata, called `<attrName>_md`), plus an additional field about the reception time of the data (`recvTime`).
+    * `csv-row`: A CSV line is added for each notified context attribute. As `json-row`, this kind of line will always contain 8 fields:
```

Comment:
It is not a valid character for the entityId, entityType, attrName and attrType, and it is replaced by '_'. Nevertheless, the values are not escaped at all. Ummm, the solution is not moving to any other field separator since any character may appear in the values. I think we could move the responsability to the user, i.e. allowing to configuring the field sepator (by default it is ',').